compiler/fir/raw-fir/raw-fir.common/src/org/jetbrains/kotlin/fir/builder/ConversionUtils.kt

```diff
@@ -73,40 +74,58 @@ fun escapedStringToCharacter(text: String): Char? {
     when (escape.length) {
         0 -> {
             // bare slash
-            return null
+            return CharacterWithDiagnostic(DiagnosticKind.IllegalEscape)
         }
         1 -> {
             // one-char escape
-            return translateEscape(escape[0]) ?: return null
+            return translateEscape(escape[0])
         }
         5 -> {
             // unicode escape
             if (escape[0] == 'u') {
                 try {
                     val intValue = Integer.valueOf(escape.substring(1), 16)
-                    return intValue.toInt().toChar()
+                    return CharacterWithDiagnostic(intValue.toInt().toChar())
                 } catch (e: NumberFormatException) {
                     // Will be reported below
                 }
             }
         }
     }
-    return null
+    return CharacterWithDiagnostic(DiagnosticKind.IllegalEscapeSequence)
 }
 
-internal fun translateEscape(c: Char): Char? =
+internal fun translateEscape(c: Char): CharacterWithDiagnostic =
     when (c) {
-        't' -> '\t'
-        'b' -> '\b'
-        'n' -> '\n'
-        'r' -> '\r'
-        '\'' -> '\''
-        '\"' -> '\"'
-        '\\' -> '\\'
-        '$' -> '$'
-        else -> null
+        't' -> CharacterWithDiagnostic('\t')
+        'b' -> CharacterWithDiagnostic('\b')
+        'n' -> CharacterWithDiagnostic('\n')
+        'r' -> CharacterWithDiagnostic('\r')
+        '\'' -> CharacterWithDiagnostic('\'')
+        '\"' -> CharacterWithDiagnostic('\"')
+        '\\' -> CharacterWithDiagnostic('\\')
+        '$' -> CharacterWithDiagnostic('$')
+        else -> CharacterWithDiagnostic(DiagnosticKind.IllegalEscapeSequence)
     }
 
+class CharacterWithDiagnostic {
+    private var diagnostic: DiagnosticKind? = null
```

Comment:
`var` isn't needed for values changed only in constructor.

Reply:
Yes, it's just because the code has been copied and transofrmed from the java source without a recheck 🤦‍♂️

Reply:
Yes, it's just because the code has been copied and transformed from the java source without a recheck 🤦‍♂️